Bumerang/app/src/main/java/com/seng480b/bumerang/DateTimePickerFragment.java

```diff
@@ -0,0 +1,128 @@
+        //maxDate is a week from today
+        Calendar maxDate = Calendar.getInstance();
+        maxDate.set(Calendar.DAY_OF_MONTH, day + 7);
+
+        DatePicker datePicker = (DatePicker) rootView.findViewById(R.id.date_picker);
+        datePicker.setMaxDate(maxDate.getTimeInMillis());
+        //minDate should be the current date (note: in order to work, it is current date minus 1 second)
+        datePicker.setMinDate(today.getTimeInMillis()-1000);
+
+        //when you tap the 'set' button
+        rootView.findViewById(R.id.date_time_set).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                int durationInMinutes = findDuration();
+                int hours = durationInMinutes/60;
+                int minutes = durationInMinutes % 60;
+                String display_text = createHourMinuteMessage(hours,minutes);
+
+                TextView time_text = (TextView) getActivity().findViewById(R.id.inputTime);
+                time_text.setText(display_text);
+
+                Intent i = new Intent();
+                i.putExtra("durationInMinutes",durationInMinutes);
+                getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, i);
+
+                getDialog().dismiss();
+            }
+        });
+
+        return rootView;
+    }
+
+    /**
+     *
+     * @return int[] returns array of 2 ints: 1rst element is hour, 2nd is minute
+     */
+    public int findDuration(){
+        DatePicker datePicker = (DatePicker) rootView.findViewById(R.id.date_picker);
+        TimePicker timePicker = (TimePicker) rootView.findViewById(R.id.time_picker);
+
+        //set up a Calendar for the current time
+        Calendar current = Calendar.getInstance();
+
+        //set up a Calendar the time the user inputs
+        Calendar changedTime = Calendar.getInstance();
+
+        int hour;
+        int minute;
+        if (Build.VERSION.SDK_INT >= 23 ) {
+            hour = timePicker.getHour();
+            minute = timePicker.getMinute();
+        } else {
+            hour = timePicker.getCurrentHour();
+            minute = timePicker.getCurrentMinute();
+        }
+
+        int day = datePicker.getDayOfMonth();
+        int month = datePicker.getMonth();
+        int year = datePicker.getYear();
+
+        changedTime.set(Calendar.HOUR_OF_DAY,hour);
+        changedTime.set(Calendar.MINUTE,minute);
+        changedTime.set(Calendar.DAY_OF_MONTH,day);
+        changedTime.set(Calendar.MONTH,month);
+        changedTime.set(Calendar.YEAR,year);
+
+        return getDiffInMinutes(changedTime,current);
+
+    }
+
+    public int getDiffInMinutes(Calendar calChanged, Calendar calCurrent){
+        long diff = calChanged.getTimeInMillis() - calCurrent.getTimeInMillis();
+        return (int)diff/(60*1000);
+    }
+
+    public String createHourMinuteMessage(int hour_diff, int minute_diff){
+        String hour_text = " hours, ";
```

Comment:
Variable names in Java should always be camel case, ie. hourText instead of hour_text.